tests/test_router_bgp.rb

```diff
@@ -229,6 +229,35 @@ def test_get_asnum
     bgp.destroy
   end
 
+  def test_nsr
+    nsr(setup_default)
+    nsr(setup_vrf)
+  end
+
+  def nsr(bgp)
+    if platform == :nexus || !@vrf[/default/]
+      refute(bgp.default_nsr,
+             'bgp nsr should be disabled by default')
+      assert_raises(Cisco::UnsupportedError) do
+        bgp.nsr = true
+        assert(bgp.nsr,
```

Comment:
You don't need the rest of these lines in the `assert_raises` block - `bgp.nsr = true` should raise the desired error immediately.